### PR TITLE
superchain github actions - circleci migration

### PR DESCRIPTION
**Pull request by raffaele-oplabs**
This PR is a complete migration from github actions to circleci. From here on github actions will be disabled.

Deleted all github actions in .github workflows.
All github actions have being migrated to circleci actions.
In circleci we are making use of dynamic configuration. There is a specific configuration file that will be used when the pipeline is triggered manually and a chain to be upload is set. This also substitute the workflow_dispatch in github.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -14,14 +14,132 @@ jobs:
...
 
 workflows:
   setup:
     jobs:
-       - setup-genesis-allocs-validation
+      - setup-genesis-allocs-validation
+
+  rollup-config-and-genesis-diff:
+    jobs:
+      - compute-config-diff-and-report-with-pr-comment:
```
Comment (geoknee) [inline .circleci/config.yml:135]: Do you want to give this job a different name to the other one?

File: .circleci/config.yml [hunk 2]
```diff
@@ -14,14 +14,132 @@ jobs:
...
+            go run <<parameters.command>>/main.go base
+      - run:
+          name: Compute diff
+          working_directory: validation
+          command: |
+            ./scripts/diff-configs.sh <<parameters.directory>>/output-base <<parameters.directory>>/output-HEAD
+          when: always
+      - run:
+          name: If command failed, create or update comment on PR
+          when: on_fail
```
Comment (geoknee) [inline .circleci/config.yml:88]: Will the entire job report as FAILED if there is a nonzero diff? We would like for the nonzero diff to definitely post a comment, but not report as failure.

**Comment by opgitgovernance** [comment]
Detected nonzero diff in genesis. This PR modifies the genesis data of at least one chain in the registry, which may or may not have been intended. You can see the detail [here](https://app.circleci.com/pipelines/workflows/f75e702e-7280-4f19-be34-76e8bff905b1).

**Comment by opgitgovernance** [comment]
Detected nonzero diff in genesis. This PR modifies the genesis data of at least one chain in the registry, which may or may not have been intended. You can see the detail [here](https://app.circleci.com/pipelines/workflows/9dc56b38-e00f-492e-b117-3d3b231ac6ee).

**Comment by opgitgovernance** [comment]
Detected nonzero diff in genesis. This PR modifies the genesis data of at least one chain in the registry, which may or may not have been intended. You can see the detail [here](https://app.circleci.com/pipelines/workflows/bddaced0-ed7a-4437-96a4-e2bc567471a0).

**Comment by opgitgovernance** [comment]
Detected nonzero diff in rollup config. This PR modifies the configuration of at least one chain in the registry, which may or may not have been intended. You can see the detail [here](https://app.circleci.com/pipelines/workflows/7df76c10-973d-473b-b5e0-ffe87efcc0c5).

**Comment by opgitgovernance** [comment]
Detected nonzero diff in rollup config. This PR modifies the configuration of at least one chain in the registry, which may or may not have been intended. You can see the detail [here](https://app.circleci.com/pipelines/workflows/d368eb97-3617-4566-a8e3-dc107bdc1d1f).

**Comment by opgitgovernance** [comment]
Detected nonzero diff in genesis. This PR modifies the genesis data of at least one chain in the registry, which may or may not have been intended. You can see the detail [here](https://app.circleci.com/pipelines/workflows/d368eb97-3617-4566-a8e3-dc107bdc1d1f).

File: .circleci/config.yml (hunk 2 above)
Comment (raffaele-oplabs) [reply .circleci/config.yml]: I fiexd this one

File: .circleci/config.yml (hunk 1 above)
Comment (raffaele-oplabs) [reply .circleci/config.yml:166]: Comment on obsolete code

**Comment by opgitgovernance** [comment]
Detected nonzero diff in rollup config. This PR modifies the configuration of at least one chain in the registry, which may or may not have been intended. You can see the detail [here](https://app.circleci.com/pipelines/workflows/bc81d8eb-df7e-466a-a1db-562041595afa/jobs/43798).

**Comment by opgitgovernance** [comment]
Detected nonzero diff in genesis. This PR modifies the genesis data of at least one chain in the registry, which may or may not have been intended. You can see the detail [here](https://app.circleci.com/pipelines/workflows/bc81d8eb-df7e-466a-a1db-562041595afa/jobs/43796).

**Comment by opgitgovernance** [comment]
Detected nonzero diff in rollup config. This PR modifies the configuration of at least one chain in the registry, which may or may not have been intended. You can see the detail [here](https://circleci.com/gh/ethereum-optimism/superchain-registry/45509).

**Comment by opgitgovernance** [comment]
Detected nonzero diff in genesis. This PR modifies the genesis data of at least one chain in the registry, which may or may not have been intended. You can see the detail [here](https://circleci.com/gh/ethereum-optimism/superchain-registry/45511).

**Comment by opgitgovernance** [comment]
Detected nonzero diff in rollup config. This PR modifies the configuration of at least one chain in the registry, which may or may not have been intended. You can see the detail [here](https://circleci.com/gh/ethereum-optimism/superchain-registry/45938).

**Comment by opgitgovernance** [comment]
Detected nonzero diff in genesis. This PR modifies the genesis data of at least one chain in the registry, which may or may not have been intended. You can see the detail [here](https://circleci.com/gh/ethereum-optimism/superchain-registry/45939).

**Comment by opgitgovernance** [comment]
Detected nonzero diff in rollup config. This PR modifies the configuration of at least one chain in the registry, which may or may not have been intended. You can see the detail [here](https://circleci.com/gh/ethereum-optimism/superchain-registry/46063).

**Comment by opgitgovernance** [comment]
Detected nonzero diff in genesis. This PR modifies the genesis data of at least one chain in the registry, which may or may not have been intended. You can see the detail [here](https://circleci.com/gh/ethereum-optimism/superchain-registry/46064).

**Comment by opgitgovernance** [comment]
Detected nonzero diff in rollup config. This PR modifies the configuration of at least one chain in the registry, which may or may not have been intended. You can see the detail [here](https://circleci.com/gh/ethereum-optimism/superchain-registry/46496).

**Comment by opgitgovernance** [comment]
Detected nonzero diff in genesis. This PR modifies the genesis data of at least one chain in the registry, which may or may not have been intended. You can see the detail [here](https://circleci.com/gh/ethereum-optimism/superchain-registry/46495).

**Comment by opgitgovernance** [comment]
Detected nonzero diff in rollup config. This PR modifies the configuration of at least one chain in the registry, which may or may not have been intended. You can see the detail [here](https://circleci.com/gh/ethereum-optimism/superchain-registry/46613).

**Comment by opgitgovernance** [comment]
Detected nonzero diff in genesis. This PR modifies the genesis data of at least one chain in the registry, which may or may not have been intended. You can see the detail [here](https://circleci.com/gh/ethereum-optimism/superchain-registry/46614).

File: .circleci/config.yml [hunk 3]
```diff
@@ -28,14 +33,24 @@ jobs:
...
+              # This will always be selected unless the user chooses to interact with the configuration by setting a non default value in the chain parameter. This would happens if the user launch the pipeline with a specific chain value via either curl or the CircleCI GUI.
+
+              echo "Generating test configuration..."
+              sh validation/genesis/validation-inputs/generate-test-config.sh
+              cp .circleci/continue_config.yml .circleci/selected_continue_config.yml
+            else
+              # This only runs if launched with curl or the CircleCI GUI with non default chain value
+
+              echo "Generating interactive configuration..."
+              cp .circleci/continue_interactive.yml .circleci/selected_continue_config.yml
```
Comment (geoknee) [inline .circleci/config.yml:48]: Nit: how do you feel about renaming `continue_config.yml` to `main_config.yml` and `continue_interactive.yml` to `artifact_upload_config.yml`? We could keep the `selected_continue_config.yml` filename, I like that.

Comment (geoknee) [reply .circleci/config.yml:48]: Also, can we rename this job name? Probably outside the scope of the PR but worth doing.

`setup-genesis-allocs-validation` => `generate-and-select-continuation-config`

**Review by geoknee — APPROVED** [review]
👏 

Tested this extensively (manually) on a sync call with @raffaele-oplabs today.

File: .circleci/config.yml (hunk 3 above)
Comment (geoknee) [reply .circleci/config.yml:48]: One final thing, can we please update the PR description (for posterity) with a brief description of what changed?

Comment (raffaele-oplabs) [reply .circleci/config.yml:48]: regarding the renaming, does generate-test-config.sh do anything with any file? I haven't renamed that for this reason. I can rename if nothing breaks there.

Comment (geoknee) [reply .circleci/config.yml:48]: Good point, that script would need to be updated. We can do this in a follow up PR if you like, and can test that rename in isolation.

**Pull request merged**